src/config/config.ts

```diff
@@ -0,0 +1,59 @@
+import { Configuration, ConfigurationParameters } from '../api';
+import { ImmutableXConfiguration, EthConfiguration } from '../types';
+import { version } from '../../package.json';
+
+const defaultHeaders = { 'x-sdk-version': `imx-core-sdk-ts-${version}` };
+
+export interface Environment extends EthConfiguration {
+  basePath: string;
+  headers?: Record<string, string>;
+}
+
+const createConfig = ({
+  coreContractAddress,
+  registrationContractAddress,
+  chainID,
+  basePath,
+  headers,
+}: Environment): ImmutableXConfiguration => {
+  if (!basePath.trim()) {
+    throw Error('basePath can not be empty');
+  }
+
+  headers = { ...(headers || {}), ...defaultHeaders };
+  const apiConfigOptions: ConfigurationParameters = {
+    basePath,
+    baseOptions: { headers },
+  };
+
+  return {
+    apiConfiguration: new Configuration(apiConfigOptions),
+    ethConfiguration: {
+      coreContractAddress,
+      registrationContractAddress,
+      chainID,
+    },
+  };
+};
+
+export const Config = {
+  get PRODUCTION() {
+    return createConfig({
+      basePath: 'https://api.ropsten.x.immutable.com',
```

Comment:
This seems not to be production?

Reply:
[c8d3b1e](https://github.com/immutable/imx-core-sdk/pull/155/commits/c8d3b1e77aeea953899ee5c36135c7be54ec1dc6)